presto-spi/src/main/java/io/prestosql/spi/block/Int128ArrayBlockBuilder.java

```diff
@@ -219,6 +219,17 @@ public boolean isNull(int position)
         return valueIsNull[position];
     }
 
+    @Override
+    public int compareTo(int leftPosition, int leftOffset, int leftLength, Block rightBlock, int rightPosition, int rightOffset, int rightLength)
+    {
+        checkReadablePosition(leftPosition);
+        int compare = Long.compare(getLong(leftPosition, SIZE_OF_LONG), rightBlock.getLong(rightPosition, SIZE_OF_LONG));
```

Comment:
I believe this is the first code in this class that assumes a specific encoding of 128 numbers.  In this case It would be 128bit little endian, so this would need to be documented somewhere.

BTW, the only reason we have compareTo (and equalsTo) was to make equals and comparison of VARCHAR and VARBINAY more efficient by removing the need to create a `Slice` object.  Not for this PR... but we should check if this is still a win, and if so, is there a better way to do this now that doesn't require these methods.

Reply:
> I believe this is the first code in this class that assumes a specific encoding of 128 numbers. In this case It would be 128bit little endian, so this would need to be documented somewhere.
> 
My concern here is that it's not a pure 128bit little endian encoding, but a comparison of 2 big endian signed longs where the most significant one is on the right.

> BTW, the only reason we have compareTo (and equalsTo) was to make equals and comparison of VARCHAR and VARBINAY more efficient by removing the need to create a `Slice` object. Not for this PR... but we should check if this is still a win, and if so, is there a better way to do this now that doesn't require these methods.

It's possible to skip now the compareTo by minor change in `UuidDistinctFromOperator#isDistinctFrom`
so, I will go with this solution leaving `Int128ArrayBlockBuilder` without any specific encoding assumption.